src/main/java/com/hivemq/cli/utils/MqttUtils.java

```diff
@@ -25,37 +26,45 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
 import java.util.stream.Collectors;
 
 public class MqttUtils {
 
-    private static final String CLIENT_ID_CHARSET = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    private static final @NotNull String CLIENT_ID_CHARSET =
+            "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
 
     public enum IdentifierWarning {
-        TOO_LONG, TOO_SHORT, CONTAINS_INVALID_CHAR
+        TOO_LONG,
+        TOO_SHORT,
+        CONTAINS_INVALID_CHAR
     }
 
     // Arrange the size of qos array to match the topics array size
     // The returned qos array will be filled with the default value represented by the first element in the given qos array
-    // if the sizes dont match up. Else this method throws an IllegalArgument exception
-    public static MqttQos[] arrangeQosToMatchTopics(final String[] topics, MqttQos[] qos) {
+    // if the sizes do not match up. Else this method throws an IllegalArgument exception
+    public static @NotNull MqttQos @NotNull [] arrangeQosToMatchTopics(
```

Comment:
```suggestion
    public staticMqttQos @NotNull [] arrangeQosToMatchTopics(
```

Reply:
Why, we do not want the reference to be null and we also do not want the array items to be null.

@nonnull Object [] o1;

o1 = null; // OK
o1 = new Object[1];
o1[0] = null; // NOT OK

Object @nonnull[] o2;

o2 = null; // NOT OK
o2 = new Object[1];
o2[0] = null; // OK